pkg/devfile/adapters/docker/utils/utils.go

```diff
@@ -233,6 +234,37 @@ func UpdateComponentWithSupervisord(comp *common.DevfileComponent, runCommand co
 	}
 }
 
+// CreateAndGetProjectVolume creates a project volume if absent and returns the
+// name of the created project volume
+func CreateAndGetProjectVolume(client lclient.Client, componentName string) (string, error) {
```

Comment:
Nit: Maybe we should change the name since we're not always creating a volume? `CreateProjectVolumeIfNeeded?` Or something similar

Reply:
Additionally, the only place where this function is called is in `docker/adapters/component/utils.go`, could we move this function there and also make it unexported?

Reply:
do you reckon i should move the supervisord stuff and make it unexported too since its only being used in component utils? 🤔